treeherder/etl/resultset_loader.py

```diff
@@ -182,5 +184,75 @@ def transform(self, repository):
         return self.fetch_resultset(pr_url, repository)
 
 
+class HgPushTransformer:
+    # {
+    #   "root": {
+    #     "payload": {
+    #       "pushlog_pushes": [
+    #         {
+    #           "time": 14698302460,
+    #           "push_full_json_url": "https://hg.mozilla.org/try/json-pushes?version=2&full=1&startID=136597&endID=136598",
+    #           "pushid": 136598,
+    #           "push_json_url": " https: //hg.mozilla.org/try/json-pushes?version=2&startID=136597&endID=136598",
+    #           "user": " james@hoppipolla.co.uk"
+    #         }
+    #       ],
+    #       "heads": [
+    #         "2f77bc4f354d9ba67ea5270b2fc789f4b0521287"
+    #       ],
+    #       "repo_url": "https://hg.mozilla.org/try",
+    #       "_meta": {
+    #         "sent": "2016-07-29T22:11:18.503365",
+    #         "routing_key": "try",
+    #         "serializer": "json",
+    #         "exchange": "exchange/hgpushes/v1"
+    #       }
+    #     }
+    #   }
+    # }
+
+    def __init__(self, message_body):
+        self.message_body = message_body
+        self.repo_url = message_body["payload"]["repo_url"]
+        self.branch = None
+
+    def transform(self, repository):
+        logger.info("transforming for {}".format(repository))
+        url = self.message_body["payload"]["pushlog_pushes"][0]["push_full_json_url"]
+        return self.fetch_resultset(url, repository)
+
+    def fetch_resultset(self, url, repository, sha=None):
+        try:
+            logger.info("fetching for {} {}".format(repository, url))
+            # there will only ever be one, with this url
+            push = fetch_json(url)["pushes"].values()[0]
+            commits = list(push["changesets"])
+            head_commit = commits[0]
+            logger.info("got head commit: {}".format(head_commit["node"]))
+            resultset = {
+                "revision": head_commit["node"],
+                "push_timestamp": push["date"],
+                "author": push["user"],
+            }
+
+            revisions = []
+            for commit in commits:
+                revisions.append({
+                    "comment": commit["desc"],
+                    "repository": repository,
+                    "author": commit["author"],
+                    "revision": commit["node"]
+                })
+
+            resultset["revisions"] = revisions
+            return resultset
+
+        except Exception as ex:
```

Comment:
I know this is mirroring the GitHub pulse ingestion implementation -- but should we be catching this?
It would seem a timeout fetching pushes should retry? So either we shouldn't try-catch at all, or should handle 404s vs timeouts differently? Though since we've just had a pulse message about a push I guess it's unlikely we'll get a 404 - not like when not being able to find the push for jobs.